d2graph/d2graph.go

```diff
@@ -1160,15 +1148,36 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 			sideLength := math.Max(obj.Width+paddingX, obj.Height+paddingY)
 			obj.Width = sideLength
 			obj.Height = sideLength
-
-		default:
+		} else {
 			if desiredWidth == 0 {
 				obj.Width += float64(paddingX)
 			}
 			if desiredHeight == 0 {
 				obj.Height += float64(paddingY)
 			}
 		}
+
+		// give shapes with icons extra padding
+		if obj.Attributes.Icon != nil {
+			paddingX += 20
+			paddingY += 20
+		}
+		switch shapeType {
+		case shape.TABLE_TYPE, shape.CLASS_TYPE, shape.CODE_TYPE, shape.IMAGE_TYPE:
+		default:
+			if obj.Attributes.Link != "" {
+				paddingX += 32
+			}
+			if obj.Attributes.Tooltip != "" {
+				paddingX += 32
+			}
+		}
+
+		if desiredWidth == 0 && desiredHeight == 0 {
```

Comment:
weren't they not grouped before? e.g. u can specify just the width or just the height. but this new code doesn't take that into account?